### PR TITLE
Real errors

### PR DESCRIPTION
**Pull request by joeybaker**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
+'use strict';
+
 var request = require('request')
   , qs = require('querystring')
   , jsonreq = request.defaults({json:true})
-  ;
 
-var chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+function makeError(err, resp){
+  var errObject = new Error(resp.statusCode + ' ' + err.reason)
```
Comment (ben-ng) [inline index.js:8]: the assumes that error is both not `null`, and an `Error`, which looking at the rest of the code, isn't necessarily true.

File: index.js [hunk 2]
```diff
@@ -27,9 +31,7 @@ Couch.prototype.get = function (id, cb) {
   request({url:this.url+encodeURIComponent(id), json:true}, function (err, resp, doc) {
     if (err) return cb(err)
     if (resp.statusCode !== 200) {
-      var e = doc ? doc : new Error('CouchDB error.') // make this smarter later
-      e.statusCode = resp.statusCode
-      return cb(e)
+      return cb(makeError(doc, resp))
```
Comment (ben-ng) [inline index.js:34]: such as here. Why is `doc` being passed as the `err` arg?

Comment (joeybaker) [inline index.js:34]: because if couch can't find something, it still returns JSON

File: index.js (hunk 1 above)
Comment (ben-ng) [inline index.js:8]: oh doh, its being passed the response `body`. nvm then.

**Comment by ben-ng** [comment]
LGTM.

**Comment by joeybaker** [comment]
thanks!

**Pull request merged**